### PR TITLE
Make PlutusType deriving more ergonomic

### PR DESCRIPTION
**Pull request by MangoIV**
### Checklist
- [x] change the underlying classes and type families
- [ ] fix ambiguity errors

### Needs
- PR #486

### CODE REVIEW

File: Plutarch.hs [hunk 1]
```diff
@@ -22,7 +22,7 @@ module Plutarch (
   PP.PMatch,
   PO.printTerm,
   PO.printScript,
-  (PL.#$),
+  -- (PL.#$), FIXME? wtf?
```
Comment (L-as) [inline Plutarch.hs:25]: elaborate?

File: Plutarch/Bool.hs [hunk 2]
```diff
@@ -126,19 +126,19 @@ x #|| y = pforce $ por # x # pdelay y
...
 
 -- | Hoisted, Plutarch level, lazily evaluated boolean or function.
 por :: Term s (PBool :--> PDelayed PBool :--> PDelayed PBool)
-por = phoistAcyclic $ plam $ \x -> pif' # x # (phoistAcyclic $ pdelay $ pcon PTrue)
+por = phoistAcyclic $ plam $ \x -> pif' # x # phoistAcyclic (pdelay $ pcon PTrue)
 
 -- | Hoisted, Plutarch level, strictly evaluated boolean or function.
 por' :: Term s (PBool :--> PBool :--> PBool)
-por' = phoistAcyclic $ plam $ \x -> pif' # x # (pcon PTrue)
+por' = phoistAcyclic $ plam $ \x -> pif' # x # pcon PTrue
```
Comment (L-as) [inline Plutarch/Bool.hs:141]: ?

File: Plutarch/Internal/PlutusType.hs [hunk 3]
```diff
@@ -1,7 +1,8 @@
-{-# LANGUAGE AllowAmbiguousTypes #-}
```
Comment (L-as) [inline Plutarch/Internal/PlutusType.hs:1]: I assume you'll undo this

File: Plutarch.hs (hunk 1 above)
Comment (MangoIV) [reply Plutarch.hs:25]: It somehow showed the error to be on that line which was just very weird as it's an ambiguity error that affects the whole module :)

File: Plutarch/Internal/PlutusType.hs (hunk 3 above)
Comment (MangoIV) [reply Plutarch/Internal/PlutusType.hs:1]: If I don't need it after my changes that'd be neat I guess. I'll see

**Comment by L-as** [comment]
This work needs to be redone after we move to 2.0. The correct fix here is to fix GHC.

**Pull request closed without merging**